codebuild/bin/test_exec_leak.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+
+set -e
+
+source codebuild/bin/s2n_setup_env.sh
+source codebuild/bin/jobs.sh
+
+function build() {
+  echo "=== BUILDING $1 ==="
+  cmake . -B$1 -DCMAKE_PREFIX_PATH=$TARGET_LIBCRYPTO_PATH ${@:2}
+  cmake --build $1 -- -j $JOBS
+}
+
+function fail() {
+    echo "test failure: $1"
+    exit 1
+}
+
+function write_exec_app() {
+cat <<EOF > build/detect_exec_leak.c
+#include <s2n.h>
+#include "unistd.h"
+
+int main() {
+    s2n_init();
+    execl("build/bin/detect_exec_leak_finish", "", NULL);
+    return 0;
+}
+EOF
+}
+
+function write_exec_finish_app() {
+cat <<EOF > build/detect_exec_leak_finish.c
```

Comment:
perhaps more of a matter of opinion, but rather than generating code from a shell script, can we just check the source code into the repo? Also instead of manually driving the compiler, could we build the executables through cmake?

Reply:
I tried placing the file in tests/unit but that causes issues (check the Testing section in description). I think this is a fine pattern since we are already doing something like this in the interning test https://github.com/aws/s2n-tls/blob/main/codebuild/bin/test_libcrypto_interning.sh

Reply:
Ya, our CMakeLists.txt glob includes everything in that directory so it would automatically try and execute it with CTest.

And I guess I makes sense to keep things consistent.